platform/ext/target/nordic_nrf/nrf5340/spm_hal.c

```diff
@@ -46,23 +60,51 @@ enum tfm_plat_err_t tfm_spm_hal_configure_default_isolation(
                   uint32_t partition_idx,
                   const struct tfm_spm_partition_platform_data_t *platform_data)
 {
-    /* FIXME we need to see how to handle this because nRF IDAU peripheral
-     * is not able to assign PRIV/nPRIV access policy configuration
+    if (!platform_data) {
+        return TFM_PLAT_ERR_INVALID_INPUT;
+    }
+
+#if defined(CONFIG_TFM_ENABLE_MEMORY_PROTECT) && (TFM_LVL != 1)
+    bool privileged = tfm_is_partition_privileged(partition_idx);
+    struct mpu_armv8m_region_cfg_t region_cfg;
+
+    if (!privileged) {
```

Comment:
In my comment I meant to do it this way:
```suggestion
    if (!tfm_is_partition_privileged(partition_idx)) {
        struct mpu_armv8m_region_cfg_t region_cfg;
```